emscripten/markerDecompress.c

```diff
@@ -96,24 +96,40 @@ void extractDataAndSave(const char* str, const char* name){
     int iset_final_index = 9;
 
     char *fsetInitialIndex = strstr(str, "\",\"fset\":\"");
+    if (fsetInitialIndex == NULL) {
+        ARLOGe("Error: 'fset' not found in the string.\n");
+        exit(EXIT_FAILURE);
+    }
     int fset_initial_index = (fsetInitialIndex - str);
 
     int fset_final_index = (fset_initial_index + 10);
 
     int iset_content_size = fset_initial_index - iset_final_index;
 
     char *fset3InitialIndex = strstr(str, "\",\"fset3\":\"");
+    if (fset3InitialIndex == NULL) {
+        ARLOGe("Error: 'fset3' not found in the string.\n");
+        exit(EXIT_FAILURE);
+    }
     int fset3_initial_index = (fset3InitialIndex - str);
     int fset3_final_index = (fset3_initial_index + 11);
 
     int fset_content_size = fset3_initial_index - fset_final_index;
 
     char *endOfStr = strstr(str, "\"}");
+    if (endOfStr == NULL) {
+        ARLOGe("Error: end of string not found.\n");
+        exit(EXIT_FAILURE);
+    }
     int endPos = endOfStr - str;
-    
+
     int fset3_content_size = endPos - fset3_final_index;
 
     // ---ISET---
+    if (iset_content_size <= 0) {
```

Comment:
same as above, this check the consistence of the string size if negative that means a corrupted file for sure....